src/main/java/fi/otavanopisto/kuntaapi/server/debug/TimedInterceptor.java

```diff
@@ -0,0 +1,60 @@
+package fi.otavanopisto.kuntaapi.server.debug;
+
+import java.io.Serializable;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+import org.apache.commons.lang3.StringUtils;
+
+@Interceptor
+@Timed
+public class TimedInterceptor implements Serializable {
+  
+  private static final long serialVersionUID = 5589726291615678816L;
+  
+  private static final String METHOD_THRESHOLD_EXCEEDED_MESSAGE = "Method %s invokation took %d ms";
+  
+  @Inject
+  private Logger logger;
+
+  @AroundInvoke
+  public Object aroundInvoke(InvocationContext invocationContext) throws Exception {
+    Method method = invocationContext.getMethod();
+    Timed timed = method.getAnnotation(Timed.class);
+    long startTime = System.currentTimeMillis();
+    try {
+      return invocationContext.proceed();
+    } finally {
+      long duration = System.currentTimeMillis() - startTime;
+      String methodDescription = getMethodDescription(method, invocationContext.getParameters());
+      
+      if (duration > timed.severeThreshold()) {
+        logger.severe(String.format(METHOD_THRESHOLD_EXCEEDED_MESSAGE, methodDescription, duration));
+      } else if (duration > timed.warningThreshold()) {
+        logger.warning(String.format(METHOD_THRESHOLD_EXCEEDED_MESSAGE, methodDescription, duration));
+      } else if (duration > timed.infoThreshold()) {
+        logger.info(String.format(METHOD_THRESHOLD_EXCEEDED_MESSAGE, methodDescription, duration));
+      }
+    }
+  }
+  
+  private String getMethodDescription(Method method, Object[] parameters) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getMethodDescription" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)